mlir/lib/Dialect/MIOpen/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -261,6 +262,295 @@ static TransformingForOp createLdsStoreLoop(OpBuilder &b, Location loc,
   return loop;
 }
 
+/// Temporary function to remove the kPack transformation currently being
+/// introduced in conv-to-gemm. kPack is a LDS-specific transformation and
+/// should not have been applied to the gridwise gemm argument. Future work will
+/// make the conversion stop doing that, and we'll be able to get rid of this.
+static Value unKpack(OpBuilder &b, Value matrix) {
+  ArrayRef<int64_t> shape = matrix.getType().cast<MemRefType>().getShape();
+  if (shape.size() != 4)
+    return matrix;
+  BottomUpTMBuilder unkpack(b, {"g", "k", "d", "kpack"}, shape,
+                            matrix.getLoc());
+  unkpack.passThrough({"g", "m"});
+  unkpack.merge("k", 1, {"k", "kpack"});
+  TransformMapAttr unkpackAttr = unkpack.get();
+  return b.create<TransformOp>(matrix.getLoc(), matrix, unkpackAttr);
+}
+
+/// Given a G x K x D matrix and the block tuning parameters and how much data
+/// each thread will load,
+/// return the dimension in which the load of this matrix from global memory
+/// should be vectorized and the length of that vector load. Also takes
+/// `tiebreaker`, the vectorization dimension to be used when both choises are
+/// equal, which should be the vectorization dimension of the store to LDS.
```

Comment:
The tiebreak is based on the LDS layout. If we can't decide based on vectorization length, we want to vectorize using the dimension that matches the fastest-moving LDS dimension, which is m/n if kpack is 1 or k otherwise.